### ISSUE TITLE
Some questions on calculating sharpness minima

### ISSUE THREAD

**Issue by JwFbupt** [henryqin1997/Minima-Sharpness-of-optimizers#1]
Hi~ I'm very delighted on your implementation of sharpness minima.
I would like to evaluate the model on  another task.
But I have some problems:
i) Is your implementation based on the paper "ON LARGE-BATCH TRAINING FOR DEEP LEARNING:
GENERALIZATION GAP AND SHARP MINIMA"?
ii) How could I transfer your implementation on my task?

Maybe the question is a little bit stupid, as reading the code without any README is actually hard.

I would appreciate if you can answer my question or give me some advices. 
YOU CAN DIRECTLY EMAIL ME  jwfan@bupt.edu.cn

**Comment by henryqin1997** [comment]
Hi, thank you for your interest in my work. Sorry for no READEME yet, this is my working directory haha.
i) the implementation is related to the paper "ON LARGE-BATCH TRAINING FOR DEEP LEARNING:
GENERALIZATION GAP AND SHARP MINIMA", but I am plotting the loss (cross-section of the minima) instead of the sharpness value, as we found it is somehow misleading using original formula. It is also stated in other papers. If you want to use the original formula, you will need to change the code in sharpness part.
ii) I modified optimizer to get the Cepsilon box, that's the most important change. And gradient accumulation is required, to run on single device.
